egs/librispeech/asr/simple_v1/local/arpa2fst.py

```diff
@@ -51,22 +64,28 @@ class Ngram(NamedTuple):
 def parse_arpa_file(filename, to_natural_log=True):
     ngrams = []
     with open(filename) as f:
-        stage = 0   # stage 0 for header, 1 for unigram, 2 for bigram, ...
+        stage = 0  # stage 0 for header, 1 for unigram, 2 for bigram, ...
         for line in f:
             line = re.sub(r'\s+$', '', line)
-            if re.match(r'^\s*$', line) or re.match(r'\\(data|end)\\', line) or re.match(r'ngram \d+=', line):
+            if re.match(r'^\s*$', line) or re.match(
```

Comment:
I would suggest using `re.compile` to create an object and using the object for matching,
which is more efficient when the ARPA file contains lots of lines.

Reply:
@jimbozhang it would be appreciated if you can fix these issue, many thanks

Reply:
OK. I'll do it later. It is said that Perl programers are not used to compile the regular expressions. 😄